polaris-react/src/components/IndexTable/components/Row/Row.tsx

```diff
@@ -120,13 +120,14 @@ export const Row = memo(function Row({
     disabled && styles['TableRow-disabled'],
     tone && styles[variationName('tone', tone)],
     !selectable &&
+      !onClick &&
       !primaryLinkElement.current &&
       styles['TableRow-unclickable'],
   );
 
   let handleRowClick;
 
-  if ((!disabled && selectable) || primaryLinkElement.current) {
+  if ((!disabled && selectable) || onClick || primaryLinkElement.current) {
```

Comment:
I was hoping we could simplify this to basically only check `disabled`, all the extra boolean logic is confusing and could lead to weird edge cases, thoughts?